overwatch/src/services/state.rs

```diff
@@ -280,21 +285,21 @@ mod test {
 
     #[async_trait]
     impl StateOperator for PanicOnGreaterThanTen {
-        type StateInput = UsizeCounter;
+        type State = UsizeCounter;
         type Settings = ();
         type LoadError = Infallible;
 
         fn try_load(
-            _settings: &<Self::StateInput as ServiceState>::Settings,
-        ) -> Result<Option<Self::StateInput>, Self::LoadError> {
+            _settings: &<Self::State as ServiceState>::Settings,
+        ) -> Result<Option<Self::State>, Self::LoadError> {
             Ok(None)
         }
 
-        fn from_settings(_settings: <Self::StateInput as ServiceState>::Settings) -> Self {
+        fn from_settings(_settings: <Self::State as ServiceState>::Settings) -> Self {
```

Comment:
Because this is a test and we know the concrete types, this settings here could be the concrete type we know to be used. If we use anything else, the compiler will complain.

Reply:
Ah, I see this now. So, we keep the settings in `ServiceState` and use it in the operator as we already know the type.

Reply:
@AlejandroCabeza, let's keep it like that. Antonio has a point. The renaming if fine though.

Reply:
@ntn-x2 The only thing is that in `StateOperator` you would have to bind the `State` type  to a bounded type. Otherwise its not constraint in the trait itself.

Reply:
Sorry, what is the goal we are trying to achieve here? Maybe I am missing that.